node/test/circuits.js

```diff
@@ -417,6 +417,74 @@ RelayNetwork.test('recovers after five successes', aliceAndBob, function t(netwo
     }
 });
 
+RelayNetwork.test('does not recover when successes are non-consecutive', aliceAndBob, function t(network, assert) {
+
+    var count = 0;
+
+    var circuit = network.getCircuit(0, 'alice', 'bob', 'call');
+
+    network.register('call', function (req, res) {
+        count++;
+        if (count > 20 && count % 4 !== 0) {
```

Comment:
Note that in this case, 1 in 4 fail after the circuit is forced into the unhealthy state.